azure-iot-device/tests/iothub/auth/test_iotedge_authentication_provider.py

```diff
@@ -4,61 +4,318 @@
-            "{}%2Fdevices%2F{}%2Fmodules%2F{}\n".format(
-                fake_hostname, fake_device_id, fake_module_id
-            )
+from .shared_auth_tests import SharedBaseRenewableAuthenticationProviderInstantiationTests
+
+
+@pytest.fixture
+def gateway_hostname():
+    return "__FAKE_GATEWAY_HOSTNAME__"
+
+
+@pytest.fixture
+def module_generation_id():
+    return "__FAKE_MODULE_GENERATION_ID__"
+
+
+@pytest.fixture
+def workload_uri():
+    return "http://__FAKE_WORKLOAD_URI__/"
+
+
+@pytest.fixture
+def api_version():
+    return "__FAKE_API_VERSION__"
+
+
+@pytest.fixture
+def certificate():
+    return "__FAKE_CERTIFICATE__"
+
+
+@pytest.fixture
+def mock_hsm(mocker, certificate):
+    mock_hsm = mocker.patch(
+        "azure.iot.device.iothub.auth.iotedge_authentication_provider.IoTEdgeHsm"
+    ).return_value
+    mock_hsm.get_trust_bundle.return_value = certificate
+    return mock_hsm
+
+
+@pytest.fixture
+def hsm(module_id, module_generation_id, workload_uri, api_version):
+    return IoTEdgeHsm(
+        module_id=module_id,
+        module_generation_id=module_generation_id,
+        workload_uri=workload_uri,
+        api_version=api_version,
+    )
+
+
+@pytest.fixture
+def auth_provider(
+    mock_hsm,
+    hostname,
+    device_id,
+    module_id,
+    gateway_hostname,
+    module_generation_id,
+    workload_uri,
+    api_version,
+):
+    return IoTEdgeAuthenticationProvider(
+        hostname=hostname,
+        device_id=device_id,
+        module_id=module_id,
+        gateway_hostname=gateway_hostname,
+        module_generation_id=module_generation_id,
+        workload_uri=workload_uri,
+        api_version=api_version,
+    )
+
+
+#######################################
+# IoTEdgeAuthenticationProvider Tests #
+#######################################
+
+
+@pytest.mark.describe("IoTEdgeAuthenticationProvider - Instantiation")
+class TestIoTEdgeAuthenticationProviderInstantiation(
+    SharedBaseRenewableAuthenticationProviderInstantiationTests
+):
+
+    # TODO: Increase coverage by completing parent class
+
+    @pytest.mark.it("Sets the gateway_hostname parameter as an instance attribute")
+    def test_gateway_hostname(self, auth_provider, gateway_hostname):
+        assert auth_provider.gateway_hostname == gateway_hostname
+
+    @pytest.mark.it("Creates an instance of the IoTEdgeHsm")
+    def test_creates_edge_hsm(self, auth_provider, mock_hsm):
+        assert auth_provider.hsm is mock_hsm
+
+    @pytest.mark.it(
+        "Sets a certificate acquired from the IoTEdgeHsm as the ca_cert instance attribute"
+    )
+    def test_ca_cert_from_edge_hsm(self, auth_provider, mock_hsm):
+        assert auth_provider.ca_cert is mock_hsm.get_trust_bundle.return_value
+        assert mock_hsm.get_trust_bundle.call_count == 1
+
+
+# TODO: Potentially get rid of this test class depending on how the parent class is tested/refactored
+@pytest.mark.describe("IoTEdgeAuthenticationProvider - _sign()")
```

Comment:
>- _sign()") [](start = 53, length = 11)

missing dot (._sign)?   Or is this a static function?

Reply:
Yup, it's missing it. Good catch. Even if it were static, it would still need a dot.

---
In reply to: [294474131](https://github.com/Azure/azure-iot-sdk-python-preview/pull/110#discussion_r294474131) [](ancestors = 294474131)